src/main/app/controller/courts/CourtTypesController.ts

```diff
@@ -24,64 +30,79 @@ export class CourtTypesController {
     res: Response,
     updated = false,
     error = '',
-    courtCourtTypes: CourtType[] = null): Promise<void> {
-    if (!courtCourtTypes) {
-
-      const slug: string = req.params.slug as string;
-      await req.scope.cradle.api.getCourtCourtTypes(slug)
-        .then((value: CourtType[]) => courtCourtTypes = value)
-        .catch(() => error += this.getCourtTypesErrorMsg);
-
+    courtTypesAndCodes: CourtTypesAndCodes = null ): Promise<void> {
+    const slug: string = req.params.slug as string;
+    if (courtTypesAndCodes == null) {
+      await req.scope.cradle.api.getCourtTypesAndCodes(slug)
+        .then((value: CourtTypesAndCodes) => courtTypesAndCodes = value)
+        .catch(() => error += this.getCourtTypesAndCodesErrorMsg);
     }
 
-    let allCourtTypes: CourtType[] = [];
+    if(courtTypesAndCodes) {
+      if (!courtTypesAndCodes.dxCodes?.some(ot => ot.isNew === true)) {
+        this.addEmptyFormsForNewEntries(courtTypesAndCodes.dxCodes);
+      }
+    }
 
-    await req.scope.cradle.api.getCourtTypes()
-      .then((value: CourtType[]) => allCourtTypes = value)
+    let courtTypes: CourtType[] = [];
```

Comment:
If there is a problem above with getting the court types and codes, we will still send a request to get the court types. Might be better to say something like: 

let courtTypes: CourtType[] = [];
If (courtTypesAndCodes.length) { 
  // call api stuff and populate
}

Reply:
export interface CourtTypesAndCodes {
  types: CourtType[];
  gbsCode?: string;
  dxCodes?: DxCode[];
}
the courtTypesAndCodes is class that holds the two lists and abs string together so can't be a array or list